estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,67 @@
+from typing import overload
+from dateutil.relativedelta import relativedelta
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+    _order = "price desc"
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)', 'The offer price must be stricty positive.')
+    ]
+
+    price = fields.Float(string="Price")
+    status = fields.Selection(string="Status", copy=False, selection=[('accepted', 'Accepted'), ('refused', 'Refused')])
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_deadline", inverse="_inverse_deadline")
+    property_type_id = fields.Many2one('estate.property.type', related="property_id.property_type_id", store=True)
+
+    @api.depends('validity')
+    def _compute_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = record.create_date + relativedelta(days=record.validity)
+            else:
+                record.date_deadline = fields.Date.today() + relativedelta(days=record.validity)
+
+    def _inverse_deadline(self):
+        for record in self:
+            if record.date_deadline:
+                record.validity = (record.date_deadline - fields.Date.today()).days
+            else:
+                record.validity = 7
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if self.env['estate.property'].browse(vals['property_id']).best_offer:
```

Comment:
```suggestion
        for vals in vals_list:
            property_id = self.env['estate.property'].browse(vals['property_id'])
            best offer = property_id.best_offer
            expected_price = property_id.expected_price
            if best_offer:
```

Since you're using browse multiple times, it's better to store the result in a variable. 
This will help prevent multiple tiime same SQL queries.

Reply:
Yes, I had already done it in previous push i think.